crates/ruff_python_ast/src/source_code/indexer.rs

```diff
@@ -139,6 +139,21 @@ impl Indexer {
         };
         Some(self.f_string_ranges[string_range_index])
     }
+
+    /// Returns `true` if the given range includes a comment.
+    pub fn comment_in_range(&self, target: TextRange) -> bool {
+        self.comment_ranges
+            .binary_search_by(|range| {
+                if target.contains_range(*range) {
```

Comment:
Nit: can we move this method to comment ranges. It could also be useful in other placer where we don't have an indexer (formatter)